vyper/context/validation/utils.py

```diff
@@ -100,6 +101,14 @@ def types_from_Attribute(self, node):
     def types_from_BinOp(self, node):
         # binary operation: `x + y`
         types_list = get_common_types(node.left, node.right)
+
+        if (
+            isinstance(node.op, (vy_ast.Div, vy_ast.Mod))
+            and isinstance(node.right, vy_ast.Num)
+            and not node.right.value
```

Comment:
Why do a zero check like this?

Reply:
This handles cases where the left side isn't a literal value, so it wasn't caught during folding.